node/bundler.ts

```diff
@@ -106,36 +109,40 @@ const bundle = async (
   // rename the bundles to their permanent names.
   await createFinalBundles([functionBundle], distDirectory, buildID)
 
-  // Retrieving a configuration object for each function.
-  const functionsConfig = await Promise.all(
-    functions.map((func) => getFunctionConfig(func, importMap, deno, logger, featureFlags)),
-  )
-
   // Creating a hash of function names to configuration objects.
-  const functionsWithConfig = functions.reduce(
-    (acc, func, index) => ({ ...acc, [func.name]: functionsConfig[index] }),
-    {} as Record<string, FunctionConfig>,
-  )
+  const internalFunctionsWithConfig: Record<string, FunctionConfig> = {}
+  for (const func of internalFunctions) {
+    internalFunctionsWithConfig[func.name] = await getFunctionConfig(func, importMap, deno, logger, featureFlags)
+  }
+
+  const userFunctionsWithConfig: Record<string, FunctionConfig> = {}
+  for (const func of userFunctions) {
+    userFunctionsWithConfig[func.name] = await getFunctionConfig(func, importMap, deno, logger, featureFlags)
+  }
 
   // Creating a final declarations array by combining the TOML file with the
   // deploy configuration API and the in-source configuration.
-  const declarationsFromConfig = mergeDeclarations(tomlDeclarations, functionsWithConfig, deployConfig.declarations)
+  const declarationsFromConfig = mergeDeclarations(
+    tomlDeclarations,
+    userFunctionsWithConfig,
+    internalFunctionsWithConfig,
+    deployConfig.declarations,
+  )
 
   // If any declarations are autogenerated and are missing the generator field
   // add a default string.
-  const declarations = internalFunctionsPath
-    ? declarationsFromConfig.map((declaration) =>
-        addGeneratorFieldIfMissing(declaration, functions, internalFunctionsPath),
-      )
-    : declarationsFromConfig
+  const declarations = declarationsFromConfig.map((declaration) =>
+    addGeneratorFieldIfMissing(declaration, functions, internalSourceDirectory),
+  )
 
   const manifest = await writeManifest({
     bundles: [functionBundle],
     declarations,
     distDirectory,
     featureFlags,
     functions,
-    functionConfig: functionsWithConfig,
+    userFunctionConfig: userFunctionsWithConfig,
```

Comment:
I'm curious why we don't want to pass it along as a single object. Right now it is merged together in the `generateManifest` function anyway. It can become super useful to pass them separately for future purposes, but I don't see we have to do it now already?